src/coreclr/jit/gentree.h

```diff
@@ -2224,6 +2230,41 @@ struct GenTree
         return (gtOper == GT_CNS_INT) ? (gtFlags & GTF_ICON_HDL_MASK) : GTF_EMPTY;
     }
 
+#ifdef FEATURE_HW_INTRINSICS
+
+    void ClearEmbRoundingMode()
+    {
+        assert(gtOper == GT_HWINTRINSIC);
+        gtFlags &= ~GTF_HW_ER_MASK;
+    }
+    void SetEmbRoundingMode(uint8_t mode)
+    {
+        assert(gtOper == GT_HWINTRINSIC);
+        ClearEmbRoundingMode();
+        switch (mode)
+        {
+            case 0x09:
+                gtFlags |= GTF_HW_ER_TONEGATIVEINFINITY;
+                break;
+            case 0x0A:
+                gtFlags |= GTF_HW_ER_TOPOSITIVEINFINITY;
+                break;
+            case 0x0B:
+                gtFlags |= GTF_HW_ER_TOZERO;
+                break;
+            default:
+                break;
```

Comment:
should this be:
```suggestion
            case 0:
                break;
            default:
                unreached();
```
?

Reply:
The new test failures come from this change:
When we call the embedded rounding APIs in the reflection call, JIT will generate a [jump table](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsiccodegenxarch.cpp#L1054) during the emit stage, this jump table is essentially a switch case containing all the possible results under different entry values, say in this case 0\~11 (by the existing design, the jump table will iterate from 0 to the max value we set, say here 11), so even this function is supposed to accept 8\~11 only, there are case when it might take 0~7. 
Based on the given information, do we consider revert the changes, or do we want to modify the jump table generation logic a bit?

p.s. I made a mistake setting the upper bound to be 12, that will take a small fix.

Reply:
It looks okay to revert the changes, and let this function accept those unexpected values. 

For reference, by the current design, `Avx2.GatherMaskVector128` is also generating a jump table containing all the results under entry values from 0~8, while this intrinsic only expect 1,2,4,8. Seems, those unexpected value shall be blocked out on the language level, say we don't define values other than those in the System.Runtime.Intrinsics.X86.FloatRoundingMode Enum. 

And for the generated results with the unexpected values would be not "harmful" as they will be the results under the default rounding mode.

Reply:
I see, so the generic jump table code generates options for all values `[0..maxValue]` for some API-specific maximum value, but some APIs (like this one) have a much more specific, smaller set of expected values. Seems ok to leave your code as it was -- maybe document (comment) why the switch allows "unexpected" values.